Rails 5 support

benlangfeld opened the pull request.

lpradovera committed:
Rails 5 version

lpradovera committed:
Version migrations

lpradovera committed:
Remove deprecated Rails methods

lpradovera committed:
Fix typo in JS

etagwerker committed:
Add ApplicationRecord and use it

etagwerker committed:
Remove old configuration

lpradovera committed:
Merge pull request #36 from etagwerker/feature/rails-5

Feature/rails 5

benlangfeld commented:
Cc @etagwerker

bklang commented on talking_stick.gemspec:
```diff
@@ -17,8 +17,8 @@ Gem::Specification.new do |s|
   s.test_files    = `git ls-files -- {test,spec,features}/*`.split("\n")
 
-  s.add_dependency 'rails', '> 3.0', '< 5.0'
-  s.add_dependency 'jquery-rails', '> 3.0', '< 5.0'
```
Since this is a breaking change we'll need to increment the major number. I suggest making one final release of TalkingStick (compatible with Rails 3 & 4) prior to merging this.

benlangfeld replied:
Or we could make it backward compatible fairly trivially.

benlangfeld replied:
@etagwerker Do you think we can make this backward compatible? I think the only thing we need to wrap in a conditional is a shim for `before_filter`, and then loosen up the jquery-rails dependency.

etagwerker replied:
@benlangfeld I don't think it's that simple to do that. There is also migration code that is not backward compatible: 

```
# for example
class CreateTalkingStickRooms < ActiveRecord::Migration[5.1]
```

We would have to add a shim there as well considering these versions: `["4.2", "5.0", "5.1"]`

Is it really worth it though? 

I'd suggest we create a `stable-4` branch with what's in `develop`/`master` right now and we release one last version of `talking_stick`.

bklang replied:
@etagwerker It is not necessary to maintain compatibility with Rails < 5.0.  Anyone on older Rails that wants to use TalkingStick can use the older version.

> I'd suggest we create a stable-4 branch with what's in develop/master right now and we release one last version of talking_stick.

Agreed.

benlangfeld replied:
@etagwerker Which migrations use Rails 5 features? I was under the impression that a migration targeting 4.2 would also execute on 5.

etagwerker replied:
@benlangfeld Good point. I guess we can make all migrations extend from `ActiveRecord::Migration[4.2]` and define the dependency like this: 

```
s.add_dependency 'rails', '>= 4.2', '< 6.0'
```

I can work on the shim for `before_filter` | `before_action`

bklang commented on app/views/talking_stick/rooms/show.html.erb:
```diff
@@ -107,7 +107,7 @@ function isInt(value){
 $(window).on('talking_stick.new_partner', function(ev, partner) {
   $('#talking-stick-partnervideos').append(partner.videoElement);
-};
+});
```
You should probably cherry-pick this to the `develop` branch since it's unrelated to Rails 5

benlangfeld replied:
Will do over the weekend.

etagwerker committed:
Use ActiveRecord::Migration[4.2]

This is compatible with Rails 4.2, 5.0 and 5.1

etagwerker committed:
Make talking_stick compatible with Rails 4.2 and 5.0 for now

Add byebug as development dependency

etagwerker committed:
Fix problem with jquery-rails

This is necessary to make sure jquery can be required like this:

    //= require jquery

etagwerker committed:
Add doc re: requirement of Rails 4.2 or Rails 5.0

bklang committed:
Merge pull request #39 from etagwerker/feature/rails-5

Feature/rails 5

bklang commented:
@etagwerker @benlangfeld What's left before we merge this? Do we still need a shim for `before_filter`? Anything else?

etagwerker commented:
@bklang @benlangfeld I think we're good to go. We don't need a shim because `before_action` has been around since Rails 4.0 and we don't need to support `before_filter` because we dropped support for Rails < 4.2.
